### PR TITLE
Add total risk chart

### PR DESCRIPTION
**Pull request by Ron-Lavi**
https://user-images.githubusercontent.com/26363699/143744638-f48cdffb-6925-489c-b24b-08e0535d51a1.mp4

### CODE REVIEW

**Comment by Ron-Lavi** [comment]
@ShimShtein can you take a look?

**Comment by Ron-Lavi** [comment]
@ShimShtein I updated the chart's legend to links that also redirect to the insights tab
gnome-shell-screenshot-3P7ZC1
:

File: webpack/InsightsHostDetailsTab/InsightsTotalRiskChart.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+import ErrorState from '../ForemanInventoryUpload/Components/AccountList/Components/ErrorState/ErrorState';
+
+const InsightsTotalRiskCard = ({ hostDetails: { id } }) => {
+  const initialRisks = {
+    1: { value: 0, title: __('Low') },
+    2: { value: 0, title: __('Moderate') },
+    3: { value: 0, title: __('Important') },
+    4: { value: 0, title: __('Critical') },
+    total: 0,
+  };
```
Comment (amirfefer) [inline webpack/InsightsHostDetailsTab/InsightsTotalRiskChart.js:27]: looks like it a constant object, could you extract it out from the component?

File: webpack/InsightsHostDetailsTab/InsightsTotalRiskChart.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+  useEffect(() => {
+    id &&
+      dispatch(
+        get({
+          key: API_KEY,
+          url: insightsCloudUrl(`hits/${id}`),
+        })
+      );
+  }, [API_KEY, dispatch, id]);
```
Comment (amirfefer) [inline webpack/InsightsHostDetailsTab/InsightsTotalRiskChart.js:44]: nit: how about using `useAPI` instead?

File: webpack/InsightsHostDetailsTab/InsightsTotalRiskChart.js [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+          eventHandlers: {
+            onClick: onChartClick,
+          },
+        },
+      ]}
+    />
+  );
+
+  if (status === STATUS.pending) {
+    cardBody = <Spinner isSVG />;
```
Comment (amirfefer) [inline webpack/InsightsHostDetailsTab/InsightsTotalRiskChart.js:130]: do we want to keep the skeleton loading consistency on this page?
we can reuse `SkeletonLoader`

File: webpack/InsightsHostDetailsTab/InsightsTotalRiskChart.js [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+  }
+
+  if (status === STATUS.ERROR) {
+    cardBody = <ErrorState error={error} />;
+  }
```
Comment (amirfefer) [inline webpack/InsightsHostDetailsTab/InsightsTotalRiskChart.js:135]: do we want to keep the skeleton loading consistency on this page?
we can reuse `SkeletonLoader` component for this behavior

File: webpack/InsightsHostDetailsTab/InsightsTotalRiskChart.js [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+        get({
+          key: API_KEY,
+          url: insightsCloudUrl(`hits/${id}`),
+        })
+      );
+  }, [API_KEY, dispatch, id]);
+
+  // eslint-disable-next-line consistent-return
+  useEffect(() => {
+    if (!hits || !hits.length) return setTotalRisks(initialRisks);
```
Comment (amirfefer) [inline webpack/InsightsHostDetailsTab/InsightsTotalRiskChart.js:48]: do we need to set the total risk  to the  initial values if it has already been set (in line 28)

File: webpack/InsightsHostDetailsTab/InsightsTotalRiskChart.js [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+    const risks = { ...initialRisks };
+    // eslint-disable-next-line camelcase
+    hits.forEach(({ total_risk }) => {
```
Comment (amirfefer) [inline webpack/InsightsHostDetailsTab/InsightsTotalRiskChart.js:51]: nit: you can rename the prop while object destructing for avoiding the lint issue

File: webpack/InsightsHostDetailsTab/InsightsTotalRiskChart.js [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+      risks[total_risk].value += 1;
+    });
+    risks.total = hits.length;
+    setTotalRisks(risks);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [hits.length]);
```
Comment (amirfefer) [inline webpack/InsightsHostDetailsTab/InsightsTotalRiskChart.js:57]: In general, using the length as a dependency might create unexpected bugs, i.e the array can be at the same size but with different values, that's why eslint's `exhaustive-deps`  is angry :)

Instead, we can use useSelector's `shallowEqual`
```js
import { useSelector, shallowEqual } from 'react-redux';

const { hits = [] } = useSelector(state => selectAPIResponse(state, API_KEY), shallowEqual);
```

**Review by amirfefer — CHANGES_REQUESTED** [review]
Thanks @Ron-Lavi ! Looks great 👍 

I left a few inline comments

**Comment by Ron-Lavi** [comment]
An issue is remaining with the useEffect being called too often and the number of total risks calculated is wrong

In addition need to add proper loading skeleton and error state

**Comment by Ron-Lavi** [comment]
Now works as expected,
@ShimShtein @amirfefer the PR is ready for another review

https://user-images.githubusercontent.com/26363699/143743769-90b697ef-fee3-4f48-b10b-a13a7e1d8bd4.mp4

**Review by ShimShtein — CHANGES_REQUESTED** [review]

File: webpack/InsightsHostDetailsTab/InsightsTab.scss [hunk 8]
```diff
@@ -88,3 +88,75 @@
 #new_host_details_insights_tab {
   padding: 24px;
 }
+
+$first-child-fill: var(--pf-global--palette--blue-100) !important;
```
[the remark this answers is not in the document]
Comment (Ron-Lavi) [reply webpack/InsightsHostDetailsTab/InsightsTab.scss:92]: I am not sure the CSS rule can be more specific than https://github.com/theforeman/foreman_rh_cloud/pull/661/files#diff-2fe4f4562fe10e9b91389c934af9b648897009d41173e49bcd56793bc774927cR148 

I think it's somehow related to the nature of SVGs and the way Victory.js loads it that without `!important` didn't work.

[the remark this answers is not in the document]
Comment (Ron-Lavi) [reply webpack/InsightsHostDetailsTab/InsightsTab.scss:92]: yep, Victory.js puts a style tag on the element itself, so no other way to override it, e.g:
```html
<path d="M 227.128, 151.777
      h9.744000000000028
      v-9.744000000000028
      h-9.744000000000028
      z" role="presentation" shape-rendering="auto" style="fill: rgb(81, 157, 233);"></path>
```

**Comment by Ron-Lavi** [comment]
@ShimShtein I added the theme props directly to the chart and chart's legend, can you take another look?

**Comment by Ron-Lavi** [comment]
@ShimShtein there is an issue in the test with foreman_ansible: `foreman_ansible plugin requires Foreman >= 3.0 but current is 2.5.4` should we create a new branch for foreman_3_1 and pin foreman_ansible version there?

**Comment by Ron-Lavi** [comment]
@ShimShtein, just rebased, thanks for your tests setup 👍🏼

**Comment by ShimShtein** [comment]
When there is only one item to show, the checkboxes are displayed incorrectly:

image

**Comment by Ron-Lavi** [comment]
Thanks @ShimShtein, it's a known issue in PF when no rows have a checkbox the main checkbox is half-baked,
I guess that's the reason by design they want to move the main checkbox to the toolbar above

**Comment by Ron-Lavi** [comment]
Thanks @ShimShtein, it's a known issue in PF when no rows have a checkbox the main checkbox is half-baked,
I guess that's the reason by design they want to move the main checkbox to the toolbar above

**Comment by ShimShtein** [comment]
I would really like to get rid of the direct css and use `theme` property instead.
Besides that - it looks great!

File: webpack/InsightsHostDetailsTab/NewHostDetailsTab.js [hunk 9]
```diff
@@ -20,7 +20,6 @@ import {
   selectSearch,
   selectHits,
 } from '../InsightsCloudSync/Components/InsightsTable/InsightsTableSelectors';
-import './InsightsTab.scss';
```
Comment (ShimShtein) [inline webpack/InsightsHostDetailsTab/NewHostDetailsTab.js:23]: Should we also remove the scss file itself?

Comment (Ron-Lavi) [reply webpack/InsightsHostDetailsTab/NewHostDetailsTab.js:23]: it's still needed for the insights tab in the old host page, when it'll be gone we can remove these files.

**Comment by ShimShtein** [comment]
Merged, thanks @Ron-Lavi!

**Pull request merged**